include/networkit/graph/DHBGraph.hpp

```diff
@@ -0,0 +1,1991 @@
+     *
+     * @return Iterator range over the edges of the graph and their weights.
+     */
+    EdgeWeightRange edgeWeightRange() const noexcept { return EdgeWeightRange(*this); }
+
+    /**
+     * Get an iterable range over the neighbors of @a.
+     *
+     * @param u Node.
+     * @return Iterator range over the neighbors of @a u.
+     */
+    NeighborRange<false> neighborRange(node u) const {
+        assert(u < upperNodeIdBound());
+        return NeighborRange<false>(*this, u);
+    }
+
+    /**
+     * Get an iterable range over the neighbors of @a u including the edge
+     * weights.
+     *
+     * @param u Node.
+     * @return Iterator range over pairs of neighbors of @a u and corresponding
+     * edge weights.
+     */
+    NeighborWeightRange<false> weightNeighborRange(node u) const {
+        assert(isWeighted());
+        assert(u < upperNodeIdBound());
+        return NeighborWeightRange<false>(*this, u);
+    }
+
+    /**
+     * Default Graph class: Returns the index of node v in the array of outgoing edges of node u.
+     * DHB: Returns the index of node v in the neighbors of outgoing edges of node u.
+     * @param u Node
+     * @param v Node
+     * @return index of node v in the array of outgoing edges of node u.
+     * TODO: We might want to support concurrency for this function in the future.
+     */
+    index indexOfNeighbor(node u, node v) const;
+
+    /**
+     * Return the i-th (outgoing) neighbor of @a u.
+     *
+     * @param u Node.
+     * @param i index; should be in [0, degreeOut(u))
+     * @return @a i-th (outgoing) neighbor of @a u, or @c none if no such
+     * neighbor exists.
+     */
+    node getIthNeighbor(node u, index i) const;
+
+    /**
+     * Return the weight to the i-th (outgoing) neighbor of @a u.
+     *
+     * @param u Node.
+     * @param i index; should be in [0, degreeOut(u))
+     * @return @a edge weight to the i-th (outgoing) neighbor of @a u, or @c +inf if no such
+     * neighbor exists.
+     */
+    edgeweight getIthNeighborWeight(node u, index i) const;
+
+    /**
+     * Get i-th (outgoing) neighbor of @a u and the corresponding edge weight.
+     *
+     * @param u Node.
+     * @param i index; should be in [0, degreeOut(u))
+     * @return pair: i-th (outgoing) neighbor of @a u and the corresponding
+     * edge weight, or @c defaultEdgeWeight if unweighted.
+     */
+    std::pair<node, edgeweight> getIthNeighborWithWeight(node u, index i) const;
+
+    /**
+     * Get i-th (outgoing) neighbor of @a u and the corresponding edge weight.
+     *
+     * @param u Node.
+     * @param i index; should be in [0, degreeOut(u))
+     * @return pair: i-th (outgoing) neighbor of @a u and the corresponding
+     * edge weight, or @c defaultEdgeWeight if unweighted.
+     *
+     * This function is not supported for DHB. Please use getIthNeighborWithWeight(node u, index i).
+     */
+    std::pair<node, edgeweight> getIthNeighborWithWeight(Unsafe, node u, index i) const;
```

Comment:
why is this function declared twice, but once we claim that it is not supported while the other one is implemented?